Ensure that form data persists when validation fails.

iamdharmesh opened the pull request:
### Description of the Change
This PR fixes the form data reset issue by ensuring that form data persists when validation fails.
It only addresses the form reset issue for JavaScript-based submissions, as HTML form submission will be removed in version 1.7.0 (already removed in https://github.com/mailchimp/wordpress/pull/126)

> [!NOTE]
> This PR resolves the issue by updating the regex to correctly determine whether the response is an error or a success based on the class in the returned markup. Ideally, this should be handled by updating the code to return a proper JSON response, but for now, the fix is applied to the existing code, with potential improvements planned for the future.

Closes #100 

### How to test the Change
1. Enable **"Use JavaScript Support?"** from the settings page.  
2. Add a subscription form using a block, shortcode, or widget.  
3. Enter an invalid value in the form (e.g., a blank email field).  
4. Press submit and verify that:  
   - The validation error appears.  
   - The entered form data persists.  
5. Enter valid data, submit the form, and verify that:  
   - The success message appears.  
   - The form data resets.  

### Changelog Entry
> Fixed - Ensure that form data persists when validation fails.

### Credits
Props @MaxwellGarceau @iamdharmesh 

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/mailchimp/wordpress/blob/develop/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

qasumitbagthariya reviewed (commented):
### QA Update  ❌ 
-------------
@iamdharmesh 

When two Mailchimp forms (including a widget) are present on the same page, the entered data does not persist after submission.

**Steps to Reproduce:**

1. Add two Mailchimp forms (one as a form and one as a widget) on the same page.
2. Fill out the form fields in both forms.
3. Submit one of the forms.
4. Observe that the data in the other form is lost.


https://github.com/user-attachments/assets/548ce099-b6d8-4fbc-bc74-bf353b67faf4




**Testing Environment** 

<Details>

- WordPress: 6.7.2
- Theme: Storefront 4.6.1
- Theme: Twenty Twenty-Four 1.3
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: fix/100

</Details>

iamdharmesh commented:
**Note:** We need to make a minor change by adding a class (similar to what we added in the `mailchimp_sf_signup_form` function in this PR) to the markup file of the updated block (https://github.com/mailchimp/wordpress/pull/126) once either of these PRs is merged into the develop branch.

iamdharmesh commented:
>Note: We need to make a minor change by adding a class (similar to what we added in the mailchimp_sf_signup_form function in this PR) to the markup file of the updated block (https://github.com/mailchimp/wordpress/pull/126) once either of these PRs is merged into the develop branch.

Added in https://github.com/mailchimp/wordpress/commit/dd812f2b922211de3e9b44cda14323d002b8f794

qasumitbagthariya commented:
Regression / Smoke Test Report  ✅ 
-------

Tested with the `smoke-testing` branch, it works as expected, similar to the fix-specific branch.

**Testing Environment** 

<Details>

- WordPress: 6.7.2
- Theme: Storefront 4.6.0
- Theme: Twenty Twenty-Four 1.3
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: smoke-testing

</Details>


Next Step- Ready to Merge 🚀
